src/test/java/org/folio/circulation/domain/OverduePeriodCalculatorServiceTest.java

```diff
@@ -0,0 +1,200 @@
+package org.folio.circulation.domain;
+
+import api.support.builders.LoanBuilder;
+import api.support.builders.LoanPolicyBuilder;
+import api.support.builders.OverdueFinePolicyBuilder;
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.apache.commons.lang3.ObjectUtils;
+import org.folio.circulation.domain.policy.LoanPolicy;
+import org.folio.circulation.domain.policy.OverdueFinePolicy;
+import org.folio.circulation.domain.policy.Period;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.LocalDate;
+import org.joda.time.LocalTime;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static api.support.fixtures.OpeningHourExamples.afternoon;
+import static api.support.fixtures.OpeningHourExamples.allDay;
+import static api.support.fixtures.OpeningHourExamples.morning;
+import static org.folio.circulation.domain.OpeningDay.createOpeningDay;
+import static org.joda.time.DateTimeConstants.MINUTES_PER_DAY;
+import static org.joda.time.DateTimeConstants.MINUTES_PER_HOUR;
+import static org.joda.time.DateTimeConstants.MINUTES_PER_WEEK;
+import static org.junit.Assert.*;
+
+@RunWith(JUnitParamsRunner.class)
+public class OverduePeriodCalculatorServiceTest {
+  private static final OverduePeriodCalculatorService calculator =
+    new OverduePeriodCalculatorService(null);
+
+  @Test
+  @Parameters
+  public void preconditionsTest(DateTime systemTime, Loan loan, boolean expectedResult) {
```

Comment:
Personally, I think it would be clearer to express these rules as individual tests that explicitly define the scenarios.

I think of parameterised tests as a way to express multiple very similar examples that exercise the same behaviour. To me, the four sets of data defined here reflect distinct and separate behaviour.

Something like:
* no fine is due when loan is not overdue
* no fine is due when loan has no due date (when can this occur?)
* no fine is due when the count closed is null
* fine is due when loan is overdue and count closed is not null

What does the count closed mean?

What do you think?

Reply:
Ok, makes sense.

Count closed indicates whether we should include closing days/hours/minutes into the calculation of overdue period. If this flag is true, we just assume that every overdue day is 24 * 60 minutes long. If it's false, we have to check the library calendar to see how many hours/minutes the library was **actually open** during the overdue period .

Reply:
> Ok, makes sense.
> 
> Count closed indicates whether we should include closing days/hours/minutes into the calculation of overdue period. If this flag is true, we just assume that every overdue day is 24 * 60 minutes long. If it's false, we have to check the library calendar to see how many hours/minutes the library was **actually open** during the overdue period .

Ok, so it should be either true to include all days or false to only include days the service point was open? Does that mean that if this value is not defined the policy cannot be applied (and is effectively invalid)?
 
> > no fine is due when loan has no due date (when can this occur?)
>
> Normally, it shouldn't. But since `dueDate` is nullable, I think we should make this check.

Ok, that is fine. I think we need to decide if that should not apply the fine, or if the system should give an error because there is insufficient information to decide.

Reply:
@OleksandrVidinieiev  Thank you for the explanation of count closed.

Maybe the code could describe that more explicitly, e.g. includeDaysWhenServicePointIsClosed?

Reply:
> Does that mean that if this value is not defined the policy cannot be applied (and is effectively invalid)?

Yes. In terms of this functionality an overdue policy is useless (effectively invalid) without `countClosed`. So maybe it should not be nullable in the first place...

Reply:
> Maybe the code could describe that more explicitly, e.g. includeDaysWhenServicePointIsClosed?

done